### PR TITLE
Replace boost by C++11 std equivalents; boost unit_test_framework remains

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

**Comment by hernando** [comment]
Have you considered *using namespace std::placeholders*?

**Comment by tribal-tec** [comment]
Not really, but might be the better solution. I'm kind of allergic to using namespace.

File: zeq/subscriber.cpp [hunk 1]
```diff
@@ -51,7 +49,7 @@ class Subscriber
 
     ~Subscriber()
     {
-        BOOST_FOREACH( SocketType socket, _subscribers )
+        for( const SocketType& socket : _subscribers )
```
Comment (eile) [inline zeq/subscriber.cpp]: const ref for an enum?

**Comment by eile** [comment]
+1

File: zeq/subscriber.cpp (hunk 1 above)
Comment (tribal-tec) [inline zeq/subscriber.cpp]: typedef std::pair< std::string, void* > SocketType;

Comment (eile) [inline zeq/subscriber.cpp]: ok, strange name then. SocketInfo? hmm, not so much better.

Comment (hernando) [inline zeq/subscriber.cpp]: If this is the key pair type used in a map, I think it's preferable to use MapType::mapped_type (not sure about the spelling) or even auto (I think auto works here)

**Comment by tribal-tec** [comment]
Updated

**Comment by tribal-tec** [comment]
ping

**Comment by tribal-tec** [comment]
ping

**Comment by tribal-tec** [comment]
retest this please

**Comment by tribal-tec** [comment]
^ was about to test the build status update of the PR. seems to work :)

**Comment by tribal-tec** [comment]
retest this please

**Comment by bilgili** [comment]
+2

**Pull request merged**